src/core/Table/columns/actionColumn.tsx

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { HeaderProps } from 'react-table';
+import { Checkbox } from '../../Checkbox';
+import SvgColumnManager from '@itwin/itwinui-icons-react/cjs/icons/ColumnManager';
+import { DropdownMenu } from '../../DropdownMenu';
+import { IconButton } from '../../Buttons/IconButton';
+import { MenuItem } from '../../Menu';
+import { TABLE_RESIZE_START_ACTION } from '../actionHandlers';
+
+const ACTION_CELL_ID = 'iui-table-action';
+
+export const ActionColumn = <T extends Record<string, unknown>>(
+  hasColumnManager: boolean,
+) => {
+  return {
+    id: ACTION_CELL_ID,
+    disableResizing: true,
+    disableGroupBy: true,
+    minWidth: 48,
+    width: 48,
+    maxWidth: 48,
+    columnClassName: 'iui-slot',
+    cellClassName: 'iui-slot',
+    disableReordering: true,
+    Header: ({ allColumns, dispatch }: HeaderProps<T>) => {
+      const [isOpen, setIsOpen] = React.useState(false);
+      if (!hasColumnManager) {
+        return null;
+      } else {
+        const headerCheckBoxes = () =>
+          allColumns
+            //Filters out any default columns made such as selection and expansion
+            .filter(({ id }) => !id.includes('iui-table'))
+            .map((column) => {
+              const { onChange, checked } = column.getToggleHiddenProps();
+              return (
+                <MenuItem
+                  key={column.id}
+                  icon={
+                    <Checkbox
+                      id={`iui-column-${column.id}`}
+                      checked={checked}
+                      disabled={column.disableToggleVisibility}
+                      onClick={(e) => e.stopPropagation()}
+                      onChange={(e) => {
+                        onChange(e);
+                        dispatch({ type: TABLE_RESIZE_START_ACTION });
+                      }}
+                    />
+                  }
+                  onClick={() => {
+                    column.toggleHidden(checked);
+                    dispatch({ type: TABLE_RESIZE_START_ACTION });
```

Comment:
Can you explain why this dispatch is needed here? Looks unrelated

/cc @bentleyvk

Reply:
It's to update the state of the resizer once a column's visibility has been toggled. It fixes [this](https://github.com/iTwin/iTwinUI-react/pull/564#discussion_r841305771) issue

Reply:
When you resize any of the columns, their width becomes static. Therefore when you remove a column, you need to update the width of all visible columns so this dispatch triggers this update by making columns' width dynamic, saving that width and setting to static width again. It is kind of a hack that I used for Table resize with resized columns to avoid recalculating each column by myself.

Reply:
Maybe there should be a comment next to the code to prevent confusion

Reply:
Comment added [here](https://github.com/iTwin/iTwinUI-react/pull/564/files#diff-e9eae8b7f2f64123d6cf3fcdc63983c5293641819768d19dfaa47ec82399fe88R49)

Reply:
Added another comment

`                // Triggers an update to resize the widths of all visible columns `